contracts/Donations.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.10;
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Counters} from "@openzeppelin/contracts/utils/Counters.sol";
+
+import "hardhat/console.sol";
+
+/**
+ * A contract to store donations before they are transferred to the charities.
+ */
+contract Donations is ERC721, AccessControl {
+    using SafeERC20 for IERC20;
+    using Counters for Counters.Counter;
+
+    bytes32 public constant WORKER_ROLE = keccak256("WORKER_ROLE");
+
+    struct DonationParams {
+        uint256 destinationId;
+        address owner;
+        IERC20 token;
+        uint256 amount;
+    }
+
+    struct Metadata {
+        uint256 destinationId;
+        IERC20 token;
+        uint256 amount;
+        uint64 expiry;
+    }
+
+    event DonationMinted(
+        uint256 indexed id,
+        uint256 indexed destinationId,
+        bytes32 indexed groupId,
+        IERC20 token,
+        uint256 expiry,
+        uint256 amount,
+        address owner
+    );
+
+    event DonationBurned(uint256 indexed id);
+
+    event DonationsSent(
+        uint256 indexed destinationId,
+        IERC20 indexed token,
+        address indexed to,
+        uint256 amount
+    );
+
+    event TTLUpdated(uint64 ttl);
+
+    Counters.Counter private metadataId;
```

Comment:
I didn't understand well about benefits to using Counters.
Is there any good point than using simple uint256?

Reply:
The only advantage is they use `unchecked` to avoid overflow checks (since they're not necessary)
It's a very small gas difference. Especially this being for Polygon

I don't think it's very relevant

Reply:
Ups, didn't notice the version, removing it 👍